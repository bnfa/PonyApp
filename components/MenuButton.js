import React from 'react';
import {TouchableOpacity, Text, StyleSheet, View} from 'react-native';
/**
 * Simple Menu button
 * @param {string} lable Button lable
 * @param {function} onPress onPress function
 */

function MenuButton({lable, onPress}) {
  return (
    <View style={styles.container}>
      <TouchableOpacity
        activeOpacity={0.5}
        onPress={onPress}
        style={styles.appButtonContainer}>
        <Text style={styles.appButtonLable}>{lable}</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: '80%',
    margin: 8,
    alignSelf: 'center',
    zIndex: 0,
  },
  appButtonContainer: {
    elevation: 8,
    backgroundColor: '#009688',
    borderRadius: 10,
    paddingVertical: 10,
    paddingHorizontal: 12,
  },
  appButtonLable: {
    fontSize: 18,
    color: '#fff',
    fontWeight: 'bold',
    alignSelf: 'center',
    textTransform: 'uppercase',
  },
});

export default MenuButton;
