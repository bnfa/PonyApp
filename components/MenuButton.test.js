import React from 'react';
import renderer from 'react-test-renderer';
import MenuButton from './MenuButton';

test('MenuButton renders correclty', () => {
  const tree = renderer
    .create(<MenuButton lable="Start" onPress={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
