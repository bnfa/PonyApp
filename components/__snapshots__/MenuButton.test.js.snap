// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton renders correclty 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "margin": 8,
      "width": "80%",
      "zIndex": 0,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#009688",
        "borderRadius": 10,
        "elevation": 8,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      Start
    </Text>
  </View>
</View>
`;
